### PR TITLE
Goodreads Block: Fix Multiple Blocks on Same Page

### PR DESCRIPTION
**Pull request by Aurorum**
## Proposed changes:

I was testing the Goodreads block a little more, so just four small changes:

- The block should probably default in descending order (ie. most recent books first) - that's what the widget currently does, and it seems most intuitive. 
- A couple of minor code enhancements - I'm not sure that `displayPreview` is necessary, and that also removes the hardcoded check for the block preview.
- There was an error caused by the block preview being an integer whereas `goodreadsId` elsewhere is a string. It probably seems more fitting to make this an integer. 
- Ensures that the `wp_enqueue_script` handle is unique so that the block works if you insert multiple on one page.

## Jetpack product discussion

N/A.

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Enable beta blocks: `add_filter( 'jetpack_blocks_variation', function () { return 'beta'; } );`
* Add two Goodreads blocks to your site for different profiles
* Confirm that on the front-end, they both now appear rather than only the first one now appearing

Since this changes `displayPreview`, it might also be worth checking that Matt's profile appears in the block preview and that you can successfully edit the URL. 

Screenshot 2024-02-22 at 18 12 56

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/goodreads/goodreads.php [hunk 1]
```diff
@@ -33,13 +33,16 @@ function register_block() {
...
-	 * Enqueue necessary scripts and styles.
-	 */
 	Jetpack_Gutenberg::load_assets_as_required( __DIR__ );
 
-	if ( isset( $attr['link'] ) ) {
-		wp_enqueue_script( 'goodreads-block', $attr['link'], array(), JETPACK__VERSION, true );
+	if ( isset( $attr['link'] ) && isset( $attr['id'] ) ) {
+		wp_enqueue_script(
+			'jetpack-goodreads-' . $attr['id'],
+			$attr['link'],
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/goodreads/goodreads.php]: I would recommend escaping those 2 values, for extra safety.

```suggestion
			'jetpack-goodreads-' . esc_attr( $attr['id'] ),
			esc_url( $attr['link'] ),
```

Comment (Aurorum) [reply projects/plugins/jetpack/extensions/blocks/goodreads/goodreads.php]: Makes sense, thanks!

Comment (Aurorum) [reply projects/plugins/jetpack/extensions/blocks/goodreads/goodreads.php]: Actually, I just tested this, and it breaks the attribute. I've switched to `esc_url_raw` which doesn't - not an expert on this, but that should be ok?

**Review by jeherve — APPROVED** [review]
This tests well for me, let's give this a try!

**Pull request merged**